app/src/main/java/com/intervaltimer/google/activities/MainActivity.java

```diff
@@ -140,15 +142,42 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
         }
 
         Intent receivedIntent = getIntent();
+        Log.i(LONG_INTENT_TAG, "Intent in onCreate() triggered");
+        handleIntent(receivedIntent);
+    }
+
+    @Override
+    protected void onNewIntent(final Intent intent) {
+        super.onNewIntent(intent);
+        Log.i(LONG_INTENT_TAG, "Intent triggered");
+        handleIntent(intent);
+    }
+
+    public void handleIntent(final Intent receivedIntent) {
         Toast show;
         if (receivedIntent != null) {
             show = Toast.makeText(getApplicationContext(),
                     "Intent: " + receivedIntent.toString(), Toast.LENGTH_LONG);
             if (Objects.equals(receivedIntent.getAction(), Intent.ACTION_VIEW)) {
+                Log.i(LONG_INTENT_TAG, Objects.requireNonNull(receivedIntent.getData()).toString());
                 getWindow().getDecorView().post(new Runnable() {
                     @Override
                     public void run() {
-                        startWorkout();
+                        boolean isWorkout = timerService.getIsWorkout();
+                        Log.i(LONG_INTENT_TAG, "Workout status " + isWorkout);
+                        Log.i(LONG_INTENT_TAG, "Current title: " + timerService.getCurrentTitle());
+                        Log.i(LONG_INTENT_TAG, "Is paused?: " + timerService.getIsPaused());
+                        Log.i(LONG_INTENT_TAG, "Current set: " + timerService.getCurrentSet());
+                        String title = timerService.getCurrentTitle();
+                        if (!Objects.equals(title, getString(R.string.workout_headline_start_timer)) &&
```

Comment:
Did you consider to move this into a getter/member function of the timerService?